src/pt/state.jl

```diff
@@ -1,39 +1,77 @@
-
 """
 The state held in each Parallel Tempering [`Replica`](@ref). 
-This interface is only needed for variation Parallel Tempering and for 
+This interface is only needed for variational Parallel Tempering and for 
 some recorders such as [`OnlineStateRecorder`](@ref).
 """
 @informal state begin
     """ 
     $SIGNATURES 
-
     The names (each a `Symbol`) of the continuous variables in the given [`state`](@ref). 
     """
     continuous_variables(state) = @abstract 
 
-    """
+    """ 
     $SIGNATURES 
+    The names (each a `Symbol`) of the discrete (Int) variables in the given state. 
+    """
+    discrete_variables(state) = @abstract 
 
+    """
+    $SIGNATURES 
     The storage within the [`state`](@ref) of the variable of the given name, typically an `Array`.
     """
     variable(state, name::Symbol) = @abstract 
+
+    """
+    $SIGNATURES
+    Update the state's entry at symbol `name` and `index` with `value`.
+    """
+    update_state!(state, name::Symbol, index, value) = @abstract
 end
 
+
+const CONTINUOUS_VARS = Ref([])
```

Comment:
At the moment there is only the notion of a "continuous" (AbstractFloat) variable and a "discrete variable" (Integer). The underlying assumption is that you must be one of the two. Parameters that take on values in the set {0.5, 1.5, 2.5}, for example, are invalid.